src/java.base/share/classes/com/sun/crypto/provider/ChaCha20Cipher.java

```diff
@@ -1257,11 +1256,9 @@ private int authUpdate(byte[] data, int offset, int length) {
      * @param out the array to write the resulting tag into
      * @param outOff the offset to begin writing the data.
      *
-     * @throws ShortBufferException if there is insufficient room to
-     *      write the tag.
```

Comment:
From line 751, `engineUpdate` can throw `ShortBufferExc` but wraps it in a `ProviderException`. Is that case possible? If so, we should change the @throws spec to a `ProviderException` but remove it from the throws line since it is a runtime exc.

Reply:
You must mean line 651?
I removed `ShortBufferException` from line 650 and got "unreported exception ShortBufferException; must be caught or declared to be thrown" when I tried to build, so that case must be possible. I changed **@throws spec** to `ProviderException` and removed it from the throws line.

Reply:
No, I didn't mean to remove it from line 650. What you proposed in the last sentence is what I meant.